### ISSUE TITLE
binding exception to local name fails

### ISSUE THREAD

**Issue by caryan** [MatthieuDartiailh/enaml#4]
Minimal example:

``` python
from enaml.widgets.api import (
    Window, Container, PushButton
)

enamldef Main(Window):
    title = "Test Exceptions"
    Container:
        PushButton:
            text = "Push Me"
            clicked ::
                print("I was clicked!")
                try:
                    pass
                except Exception:
                    pass
                print("unnamed exception worked")
                try:
                    pass
                except Exception as e:
                    pass
                print("named exception worked")
```

works in enaml with Python 2.7 but fails in Python 3.5 with:

``` shell
cryan@cryan-Precision-5510 ~/Programming/Repos/PyQLab $ enaml-run test_exception.enaml 
Traceback (most recent call last):
  File "/home/cryan/anaconda3/bin/enaml-run", line 6, in <module>
    sys.exit(enaml.runner.main())
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/runner.py", line 47, in main
    code = EnamlCompiler.compile(ast, enaml_file)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_compiler.py", line 178, in compile
    return compiler.visit(node)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_ast.py", line 351, in visit
    result = visitor(node, *args, **kwargs)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_compiler.py", line 194, in visit_Module
    self.visit(item)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_ast.py", line 351, in visit
    result = visitor(node, *args, **kwargs)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_compiler.py", line 217, in visit_EnamlDef
    code = EnamlDefCompiler.compile(node, cg.filename)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enamldef_compiler.py", line 233, in compile
    return compiler.visit(node)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_ast.py", line 351, in visit
    result = visitor(node, *args, **kwargs)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enamldef_compiler.py", line 248, in visit_EnamlDef
    node, second_args, index_map, filename
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enamldef_compiler.py", line 185, in compile
    compiler.visit(node)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_ast.py", line 351, in visit
    result = visitor(node, *args, **kwargs)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enamldef_compiler.py", line 197, in visit_EnamlDef
    self.visit(item)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_ast.py", line 351, in visit
    result = visitor(node, *args, **kwargs)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/block_compiler.py", line 128, in visit_ChildDef
    self.visit(item)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_ast.py", line 351, in visit
    result = visitor(node, *args, **kwargs)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/block_compiler.py", line 128, in visit_ChildDef
    self.visit(item)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/enaml_ast.py", line 351, in visit
    result = visitor(node, *args, **kwargs)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/block_compiler.py", line 154, in visit_Binding
    cmn.gen_operator_binding(cg, node.expr, index, node.name)
  File "/home/cryan/anaconda3/lib/python3.5/site-packages/enaml/core/compiler_common.py", line 756, in gen_operator_binding
    code = compile(node.value.ast, cg.filename, mode=mode)
TypeError: AST identifier must be of type str
```

I've tried poking around the ast tree in the debugger and all the id's do seem to be strings so I'm not quite sure what it's failing on.

``` python
ipdb> node.value.ast.body
[<_ast.Assign object at 0x7f6eade8ccc0>, <_ast.If object at 0x7f6eade8c4a8>, <_ast.Try object at 0x7f6eade91978>]
ipdb> try_node = node.value.ast.body[2]
ipdb> except_handler = try_node.handlers[0]
ipdb> except_handler.name.id
'e'
ipdb> type(except_handler.name.id)
<class 'str'>
ipdb>
```

**Comment by MatthieuDartiailh** [comment]
Thank you for letting me know. I will see what I can do and if you come up with a solution let me know.

**Comment by caryan** [comment]
Sounds good.  Which branch is the ecpy channel built off of?

**Comment by MatthieuDartiailh** [comment]
Python3-parsing

**Comment by MatthieuDartiailh** [comment]
Actually in Python 3 the name attribute of an ExceptClause should be a string and not a Name node.
However this does not fix the whole issue as when trying to access to the exception, no exception is set.
I will push the partial fix but I don't have time to go further right now.

**Comment by MatthieuDartiailh** [comment]
Actually I found another mistake in the raising of exception, so this should be fixed. I will trigger the re-building of the ecpy channel package. Once you confirm that the issue is solved please close.

**Comment by caryan** [comment]
Thanks a million. Closed by e6d2bf5e2cc3848f65d3d858c86ffc49a6f36473 and 01d824ce99d01a75c9175318a554086aa63792f8
